src/bors/handlers/workflow.rs

```diff
@@ -77,6 +78,25 @@ pub(super) async fn handle_workflow_completed(
         branch: payload.branch,
         commit_sha: payload.commit_sha,
     };
+
+    if let Some(running_time) = payload.running_time {
+        let running_time_as_duration =
```

Comment:
The status of the workflow should be modified before we store the status into the DB (`update_workflow_status`). I would also post a comment to the PR (see `find_pr_by_build`), because this should be a rare occurrence that should be immediately visible.